openedx/core/lib/xblock_serializer/utils.py

```diff
@@ -89,6 +94,38 @@ def collect_assets_from_text(text, course_id, include_content=False):
         yield info
 
 
+def get_python_lib_zip_if_using(olx: str, course_id: CourseKey) -> StaticFile | None:
+    """
+    When python_lib is in use, capa problems that contain python code should be assumed to depend on it.
+
+    Note: for any given problem that uses python, there is no way to tell if it
+    actually uses any imports from python_lib.zip because the imports could be
+    named anything. So we just have to assume that any python problems may be
+    using python_lib.zip
+    """
+    MATCH_STRINGS = (
+        '<script type="text/python">',
+        "<script type='text/python'>",
+        '<script type="loncapa/python">'
+        "<script type='loncapa/python'>",
+    )
+    found = False
+    for check in MATCH_STRINGS:
+        if check in olx:
+            found = True
+            break
+    if found:
+        python_lib_filename = getattr(settings, 'PYTHON_LIB_FILENAME', DEFAULT_PYTHON_LIB_FILENAME)
+        asset_key = StaticContent.get_asset_key_from_path(course_id, python_lib_filename)
+        # Now, it seems like this capa problem uses python_lib.zip - but does it exist in the course?
+        asset_handle = AssetManager.find(asset_key, throw_on_not_found=True, as_stream=True)
+        if asset_handle:
+            asset_handle.close()  # We don't need to read the contents at this time.
```

Comment:
```suggestion
        if get_python_lib_zip(contentstore, course_id):
```

I wonder if it wouldn't be easier to import the existing function: https://github.com/openedx/edx-platform/blob/a389a9ff10fd06bd053107fc6a4f19c7d24f7233/xmodule/util/sandboxing.py#L35-L43

We can pass the `contentstore` from `from xmodule.contentstore.django import contentstore`.